File: backend/mpp1/Model/DTO/UserDTO.cs
using System.ComponentModel.DataAnnotations;
using mpp1.Enums;
using mpp1.Validators.ClientValidators;

namespace mpp1.Model.DTO;

public class UserDto
{
    public string Username { get; set; } = null!;
    public string Bio { get; set; } = null!;
    public string Location { get; set; } = null!;
    public DateOnly Birthday { get; set; }
    public GendersEnum Gender { get; set; }
    public MaritalStatusEnum MaritalStatus { get; set; }
    public RolesEnum Role { get; set; }
    public int NumberOfItemsPerPage { get; set; }
    public int? NumberOfClients { get; set; }
    public int? NumberOfVehicles { get; set; }
    public int? NumberOfIncidents { get; set; }
    public int? NumberOfRents { get; set; }
}

File: frontend/src/model/UserDto.ts
import { GenderEnum } from "@/enums/GenderEnum";
import { MaritalStatusEnum } from "@/enums/MaritalStatusEnum";
import { RolesEnum } from "@/enums/RolesEnum";

export interface UserDto {
  username: string;
  bio: string;
  location: string;
  birthday: string;
  gender: GenderEnum;
  role: RolesEnum;
  maritalStatus: MaritalStatusEnum;
  numberOfItemsPerPage?: number;
  numberOfClients?: number;
  numberOfVehicles?: number;
  numberOfIncidents?: number;
  numberOfRents?: number;
}
